### PR TITLE
Publish wkcuber docs to S3

### PR DESCRIPTION
**Pull request by normanrz**
### Description:
- Publishes the wkcuber docs to a bucket in S3. Example: https://static.webknossos.org/lib-docs/docs_s3/wkcuber/api.html
- Refactors CI to just run the docker builds once

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [x] Updated Changelog

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -95,6 +80,65 @@ jobs:
     - name: Remove reference magnification data
       run: rm -r testdata/tiff_mag_2_reference/
 
+    - name: Generate Docs
+      if: matrix.python-version == '3.7'
+      run: |
+        docs/api.sh --persist
+    
+    - name: Push docs (for branch)
+      if: matrix.python-version == '3.7'
```
Comment (philippotto) [inline .github/workflows/main.yml:89]: should this job check whether only a tag was pushed? or does it simply work in that case, too?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -95,6 +80,65 @@ jobs:
     - name: Remove reference magnification data
       run: rm -r testdata/tiff_mag_2_reference/
 
+    - name: Generate Docs
+      if: matrix.python-version == '3.7'
```
Comment (philippotto) [inline .github/workflows/main.yml:84]: is there a specific reason for using 3.7 here? I think I would prefer to build the docs with 3.8 if there's no counter argument I'm missing.

**Review by philippotto — APPROVED** [review]
sweet stuff :candy:

File: .github/workflows/main.yml (hunk 1 above)
Comment (normanrz) [reply .github/workflows/main.yml]: good point!

File: .github/workflows/main.yml (hunk 2 above)
Comment (normanrz) [reply .github/workflows/main.yml]: i switched to 3.8

File: README.md [hunk 3]
```diff
@@ -128,6 +128,9 @@ tests/scripts/all_tests.sh
 
 PyPi releases are automatically pushed when creating a new Git tag/Github release. 
 
+## API documentation
+Check out the [latest version of the API documentation](https://static.webknossos.org/lib-docs/master/wkcuber/api.html).
```
Comment (philippotto) [inline README.md:132]: :+1: just wanted to suggest that

Comment (hotzenklotz) [reply README.md:132]: Wrong link? (at least does not work for me) Did you mean `https://static.webknossos.org/lib-docs/docs_s3/wkcuber/api.html`?

Comment (normanrz) [reply README.md:132]: The link will work, once it is deployed on `master` branch

Comment (hotzenklotz) [reply README.md:132]: Never mind

**Pull request merged**